perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -1112,148 +1322,116 @@ public static void sendPasswordResetEmail(User user, String email, String namesp
 		message.setTo(email);
 		message.setFrom(BeansUtils.getCoreConfig().getMailchangeBackupFrom());
 
-		String instanceName = BeansUtils.getCoreConfig().getInstanceName();
+		// set subject and body
+		message.setSubject(subjectOfEmail);
+		message.setText(bodyOfEmail);
 
-		if (subject == null) {
-			message.setSubject("[" + instanceName + "] Password reset in namespace: " + namespace);
-		} else {
-			subject = subject.replace("{namespace}", namespace);
-			subject = subject.replace("{instanceName}", instanceName);
-			message.setSubject(subject);
+		// send email
+		try {
+			mailSender.send(message);
+		} catch (MailException ex) {
+			throw new InternalErrorException("Unable to send email.", ex);
```

Comment:
Please do explicit logging in error mode for that exception. Thank you.

Reply:
Added.

Reply:
Location of the error in the code should be visible from the stacktrace of the passed `ex`. I would shorten the error message like: ``Unable to send mail to "email".`` or something like this.

Reply:
Changed.